_posts/2017-07-14-Rust-Roadmap-Update.md

```diff
@@ -0,0 +1,378 @@
+
+First, a meta note. If you'd like to follow along with the progress on
+a particular roadmap initiative, or to find out more about how you can
+get involved, the best place to go is the
+[list of issues on the rust-roadmap repo](https://github.com/rust-lang/rust-roadmap/issues/).
+There you will find an issue dedicated to each of the major
+initiatives that we are pushing on. These issues contain links to
+ongoing work. You'll find a number of links to issues like this in the
+descriptions that follow.
+
+### Rust should have a lower learning curve
+
+The most direct way to make Rust easier to learn is to improve the way
+that we teach it. To that end, we've been hard at work on a brand new
+edition of the "Rust" book ([roadmap issue][rr7]), and we now have a
+[complete draft available online][book]. This new edition puts
+ownership front and center and it also has expanded coverage of a
+number of other areas in Rust, such as error handling, testing,
+matching, modules, and more. Even better, you can
+[pre-order a printed version][preorder] through No Starch Press. If
+you'd like to help out, there is still
+[lots of editing work to be done!](https://github.com/rust-lang/book/projects/1)
+
+[rr7]: https://github.com/rust-lang/rust-roadmap/issues/7
+[preorder]: https://www.nostarch.com/rust
+[book]: https://doc.rust-lang.org/book/
+
+We've also been working on a number of language changes aimed at
+improving [language ergonomics][ergo]. These range from long-standing
+proposals, like [non-lexical lifetimes][rr16] or [`impl Trait`], to
+newer ideas, like the recently approved RFCs on [trait aliases][] and
+[match ergonomics][]. On the [roadmap issue][rr17], you will find a
+large list of initiatives, organized by the part of the language that
+they target (e.g., ownership/borrowing, the trait system, etc). We
+are actively looking for people to help with writing RFCs but also
+implementing the accepted RFCs -- if you're interested in helping out
+with something, look for the "mentoring" contacts listed in the
+roadmap issue or on the tracking issue for a specific RFC. And, of
+course, if you think you've got a good idea that's not listed, open up
+a thread on internals and let's talk about it!
+
+[ergo]: https://blog.rust-lang.org/2017/03/02/lang-ergonomics.html
+[rr17]: https://github.com/rust-lang/rust-roadmap/issues/17
+[rr16]: https://github.com/rust-lang/rust-roadmap/issues/16
+[`impl Trait`]: https://github.com/rust-lang/rfcs/pull/1951
+[trait aliases]: https://github.com/rust-lang/rfcs/pull/1733
+[match ergonomics]: https://github.com/rust-lang/rfcs/pull/2005
+
+### Rust should have a pleasant edit-compile-debug cycle
+
+We've been targeting the problem of improving compiler performance in
+a number of different ways. One of the simplest is the
+[`cargo check` command that we released in Rust 1.16][1.16]
+-- this command does a limited form of compilation which skips
+code-generation and simply looks for type-errors. Since code
```

Comment:
maybe leave out the `type-`? It checks for all kinds of errors, like parser ones, and also runs all lints.